[Eugene Chua] iP

eugene3231 opened the pull request.

Commit:
Add Gradle support

eugene3231 committed:
add greet, echo and exit command

eugene3231 committed:
add add, list commands

eugene3231 committed:
added done command and Task class

eugene3231 committed:
add todo, event and deadline

eugene3231 committed:
I/O redirection testing

eugene3231 committed:
added handling exceptions

eugene3231 committed:
added delete function

eugene3231 committed:
added enum use

eugene3231 committed:
update EXPECTED.txt file

eugene3231 committed:
fix DukeException crashes, add more enums

eugene3231 committed:
catch specific exceptions, use sadface unicode

eugene3231 committed:
add save feature

eugene3231 committed:
add datetime for deadline

eugene3231 committed:
Merge branch 'branch-Level-7' into master

eugene3231 committed:
Merge branch 'branch-Level-8' into master

# Conflicts:
#	src/main/java/TaskHandler.java

eugene3231 committed:
add smiley for DukeExceptions, format list cmd better

eugene3231 committed:
add clear command

eugene3231 committed:
add commands, parser and ui classes

eugene3231 committed:
move greeting to ui

eugene3231 committed:
move classes into packages

eugene3231 committed:
remove old save file

eugene3231 committed:
Revert "remove old save file"

This reverts commit 404f02bd42137d84f84e2541e4cfa611a9616090.

eugene3231 committed:
remove new save file

eugene3231 committed:
move save data into package

eugene3231 committed:
Revert "move save data into package"

This reverts commit 2eb5e253bc84afb8829d8ba2e56193cf96598e8d.

eugene3231 committed:
move save file into package

eugene3231 committed:
allow specifying save file location in main

eugene3231 committed:
fix EOF for input

eugene3231 committed:
add showing erroneous input for exceptions

eugene3231 committed:
add handling corrupted task format in save file

eugene3231 committed:
update Duke logo

eugene3231 committed:
Add JUnit tests

eugene3231 committed:
Add JavaDoc comments

eugene3231 committed:
Add JavaDoc comments

eugene3231 committed:
Add find command

eugene3231 committed:
Merge branch 'branch-A-JavaDoc' into master

eugene3231 committed:
Merge branch 'branch-A-CodingStandard' into master

eugene3231 committed:
Merge branch 'branch-A-Level-9' into master

eugene3231 committed:
Merge branch 'branch-A-JavaDoc' into master

eugene3231 committed:
rename storage termination boolean

eugene3231 committed:
Merge remote-tracking branch 'origin/master' into master

eugene3231 committed:
update JavaDoc

eugene3231 committed:
fix lines that exceed line limit

eugene3231 committed:
fix meaningful variable names

eugene3231 committed:
test clear commit message function

eugene3231 committed:
Revert "test clear commit message function"

This reverts commit 19437cd1

eugene3231 committed:
add back duke logo

eugene3231 committed:
update imports in Ui and constant var name in storage

eugene3231 committed:
Merge branch 'add-gradle-support' of github.com:nus-cs2103-AY2021S1/ip into master

eugene3231 committed:
add gradle support

eugene3231 committed:
add checkstyle support

eugene3231 committed:
fix checkstyle errors

eugene3231 committed:
Add GUI for duke

eugene3231 committed:
Revert "Add GUI for duke"

This reverts commit 6013e930739b89fc491c825c4e5239cab62c175a.

eugene3231 committed:
Add GUI for duke

eugene3231 committed:
Merge branch 'branch-Level-10' into master

zeling595 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,201 @@
+    protected static boolean hasLoadingError = false;
+    protected static final String FILE_NAME = "/save.txt";
+    protected final String location;
+    protected final File savefile;
```
maybe savedFile is a better name?

zeling595 replied:
maybe savedFile is a better name?

zeling595 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,57 @@
+            DukeException.tryAgain();
+        }
+    }
+
```
A JavaDoc for the method will be good

zeling595 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @return String representing a task in save file.
+     */
+    public String toSaveFormat() {
```
Method name should start with a verb.

zeling595 replied:
Method name should start with a verb.

zeling595 commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,64 @@
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import org.junit.jupiter.api.Test;
```
Group relevant import statements together.

zeling595 replied:
Group relevant import statements together.

zeling595 commented on src/main/java/duke/command/DoneCommand.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
A JavaDoc for the method will help others to understand this methods better.

zeling595 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,149 @@
+        return logo + "\n" + greeting + "\n" + doForYou;
+    }
+
+    public static void setLogo() {
```
"+" should be at the start of a new line

zeling595 reviewed (commented):
Your coding standard is very good. Keep it up:))

eugene3231 committed:
Add styling for GUI

eugene3231 committed:
Add Assertions

eugene3231 committed:
Improve code using peer review comments

eugene3231 committed:
Improve code using code quality topics

eugene3231 committed:
Update checkstyle

eugene3231 committed:
Fix JUnit tests

eugene3231 committed:
Remove unused imports

eugene3231 committed:
Merge pull request #1 from eugene3231/branch-A-Assertions

Merging A-Assertions into master

eugene3231 committed:
Merge branch 'master' into branch-A-CodeQuality

eugene3231 committed:
Merge pull request #2 from eugene3231/branch-A-CodeQuality

Merging branch-A-CodeQuality into master

eugene3231 committed:
Refactor find command

eugene3231 committed:
Fix find command indentation

eugene3231 committed:
Implement help feature

eugene3231 committed:
Add undo feature

eugene3231 committed:
Merge pull request #3 from eugene3231/branch-C-Undo

Add undo feature

eugene3231 committed:
Add help command

eugene3231 committed:
Add help details for undo command

eugene3231 committed:
Improve formatting for GUI

eugene3231 committed:
Make GUI more responsive

eugene3231 committed:
Make GUI's textfield and button responsive

eugene3231 committed:
Add more assert usage

eugene3231 committed:
Update Javadoc

eugene3231 committed:
Update checkstyle version

eugene3231 committed:
Improve SLAP abstraction in Parser

eugene3231 committed:
Update README.md

eugene3231 committed:
Update Ui.png

eugene3231 committed:
Merge remote-tracking branch 'origin/master' into master

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update Ui.png

eugene3231 committed:
Set theme jekyll-theme-minimal

eugene3231 committed:
Update README.md

eugene3231 committed:
Set theme jekyll-theme-cayman

eugene3231 committed:
Set theme jekyll-theme-minimal

eugene3231 committed:
Delete _config.yml

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md (+6 squashed commit)

Squashed commit:

[29b7e2d] Update README.md

[0f107a3] Update README.md

[e47d020] Update README.md

[44310d7] Update README.md

[eac4a6f] Update README.md

[ca69755] Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md

eugene3231 committed:
Merge branch 'master' of github.com:eugene3231/ip into master

eugene3231 committed:
Update README.md

eugene3231 committed:
Update README.md
